llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -10777,7 +10777,6 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
     // If the selects are the only uses of the compares, they will be
     // dead and we can adjust the cost by removing their cost.
     if (VI && SelectOnly) {
-      assert(!Ty->isVectorTy() && "Expected only for scalar type.");
```

Comment:
Shall it be something like
```
assert((SLPReVec || !Ty->isVectorTy()) && "Expected only for scalar type.");
```
?

Reply:
I think the reason we use assert here is we don't want `GetMinMaxCost` accidentally remove the cost of `CmpInst` when the cost of vectorized code is estimated.
```
  case TreeEntry::MinMax: {
    auto GetScalarCost = [&](unsigned Idx) {
      return GetMinMaxCost(OrigScalarTy);
    };
    auto GetVectorCost = [&](InstructionCost CommonCost) {
      InstructionCost VecCost = GetMinMaxCost(VecTy);
      return VecCost + CommonCost;
    };
    return GetCostDiff(GetScalarCost, GetVectorCost);
  }
```
But `GetVectorCost` does not pass `VI` to `GetMinMaxCost`, so the assert will never be triggered.
Add `SLPReVec` here is weird because they mean different thing. It will make `GetVectorCost` accidentally minus the cost of `CmpInst` when `REVEC` is enabled. But it violates the original purpose of the assert.

Reply:
Add `SLPReVec` is better.